packages/devtools/src/schema.ts

```diff
@@ -0,0 +1,415 @@
+    Object.keys(modelDefinition.schema.$defs || {}).forEach((embedName: string) => {
+      if (definitionOccurences[embedName] === undefined) {
+        definitionOccurences[embedName] = 0
+      }
+      definitionOccurences[embedName] += 1
+    })
+  })
+
+  const commonEmbeds: Array<string> = []
+  Object.keys(definitionOccurences).forEach((embedName: string) => {
+    if (definitionOccurences[embedName] > 1) {
+      commonEmbeds.push(embedName)
+    }
+  })
+
+  return commonEmbeds
+}
+
+/** @internal */
+function embeddedObjectsDefinitionsFromGraphQLSchema(
+  schema: GraphQLSchema
+): Record<string, JSONSchema.Object> {
+  const definitions: Record<string, JSONSchema.Object> = {}
+  mapSchema(schema, {
+    [MapperKind.OBJECT_TYPE]: (objectConfig: GraphQLObjectType) => {
+      const modelDirective = getCeramicModelDirective(schema, objectConfig)
+      if (!modelDirective) {
+        definitions[objectConfig.name] = embeddedObjectDefinitionFromObjectConfig(objectConfig)
+      }
+      return objectConfig
+    },
+  })
+  return definitions
+}
+
+/** @internal */
+function modelsFromGraphQLSchema(
+  schema: GraphQLSchema,
+  definitions: Record<string, JSONSchema.Object>
+): Record<string, ModelDefinition> {
+  const models: Record<string, ModelDefinition> = {}
+  mapSchema(schema, {
+    [MapperKind.OBJECT_TYPE]: (objectConfig: GraphQLObjectType) => {
+      const modelDirective = getCeramicModelDirective(schema, objectConfig)
+      if (modelDirective) {
+        models[`${objectConfig.name}ID`] = modelFromObjectConfig(
+          modelDirective,
+          objectConfig,
+          definitions
+        )
+      }
+      return objectConfig
+    },
+  })
+
+  if (Object.keys(models).length === 0) {
+    throw new Error('No models found in Composite Definition Schema')
+  }
+
+  return models
+}
+
+/** @internal */
+function embeddedObjectDefinitionFromObjectConfig(
+  objectConfig: GraphQLObjectType
+): Record<string, any> {
+  const properties: Record<string, any> = {}
+  const required: Array<string> = []
+  for (const [fieldName, fieldDefinition] of Object.entries(objectConfig.getFields())) {
+    properties[fieldName] = fieldSchemaFromFieldDefinition(
+      fieldDefinition.name,
+      fieldDefinition.type,
+      fieldDefinition.extensions?.ceramicExtensions as Record<string, any>
+    )
+    if (fieldDefinition.type instanceof GraphQLNonNull) {
+      required.push(fieldName)
+    }
+  }
+
+  return {
+    type: 'object',
+    title: objectConfig.name,
+    properties: properties,
+    required: required.length > 0 ? required : undefined,
+  }
+}
+
+/** @internal */
+function modelFromObjectConfig(
+  modelDirective: ModelDirective,
+  objectConfig: GraphQLObjectType,
+  allDefinitions: Record<string, JSONSchema.Object>
+): ModelDefinition {
+  const modelSchema: JSONSchema.Object = {
+    $schema: 'https://json-schema.org/draft/2020-12/schema',
+    type: 'object',
```

Comment:
```suggestion
    type: 'object',
    additionalProperties: false,
```

Reply:
I fixed this, not sure why GH doesn't catch it as outdated.